### ISSUE TITLE
Modify code to call Microsoft's Azure OpenAI chat completion instead of OpenAI?

### ISSUE THREAD

**Issue by aymiee** [Azure-Samples/azure-search-openai-demo#471]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [ ] feature request
- [X] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Mention any other details that might be useful
Can someone shed some light on this for me.  On Microsoft's page: https://learn.microsoft.com/en-us/legal/cognitive-services/openai/data-privacy:

> Your fine-tuned Azure OpenAI models are available exclusively for your use. The Azure OpenAI Service is fully controlled by Microsoft; Microsoft hosts the OpenAI models in Microsoft’s Azure environment and the **Service does NOT interact with any services operated by OpenAI** (e.g. ChatGPT, or the **OpenAI API**).

Is the above what's occurring in the Azure OPenAI ChatPlayground?  What does Microsoft mean by "**Service does NOT interact with any services operated ....   (or the OpenAI API)**"?

Can we be certain that Microsoft's Azure OpenAI ChatPlayground does not call OpenAI's url (https://api.openai.com/v1)?  I wasn't able to capture the traffic from Microsoft Azure Open AI Chatplayground.  Still researching; if anyone knows, that would help also.

In this demo's codebase, it is referencing OpenAI library and calls are made to OpenAI url .  If the Azure OpenAI Playground is calling it's own implementation of ChatCompletion, can I call Microsoft's version instead?  I could easily port this code to an Azure Function, if I need to be in the Azure environment.  

In summary,  can I call Microsoft's ChatCompletion endpoint instead of OpenAI's.  

Thanks in advance.

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by pamelafox** [comment]
@aymiee This sample uses Azure's OpenAI. You can see it configured here, where it sets api_type:

https://github.com/Azure-Samples/azure-search-openai-demo/blob/0f5fb2668c52c56d9961f5dde1f2bb831c6f62ef/app/backend/app.py#L43

This whole sample is designed to be deployed entirely on Azure resources. It uses App Service as the host, for both the frontend and backend.

**Comment by aymiee** [comment]
Thank you for the clarification. I was reviewing the backend code at app/backend/approaches/chatreadretrieveread.py :

> class ChatReadRetrieveReadApproach(Approach):
>
>     """
>     Simple retrieve-then-read implementation, using the Cognitive Search and OpenAI APIs directly. It first retrieves
>     top documents from search, then constructs a prompt with them, and then uses OpenAI to generate an completion
>     (answer) with that prompt.
>     """

and 

>  #STEP 3: Generate a contextual and content specific answer using the search results and chat history
>         completion = openai.Completion.create(
>             engine=self.chatgpt_deployment, 
>             prompt=prompt, 
>             temperature=overrides.get("temperature") or 0.7, 
>             max_tokens=1024, 
>             n=1, 
>             stop=["<|im_end|>", "<|im_start|>"])



The comments: ".._using the Cognitive Search and OpenAI APIs directly..._"  and the use "openai.Completion.create"  startled me a bit.

**Comment by pamelafox** [comment]
Sorry for the confusion! It would actually be possible to use the non-Azure OpenAI API with this sample with a little configuration change, and there's a PR for adding that as an option. The important aspect is openapi.api_type - always check that if you're not sure. Plus, the keys/credentials aren't shared across the two, so you wouldnt be able to use OpenAI's version with an Azure key.
